Makefile

```diff
@@ -51,16 +53,29 @@ default: build
 	@echo "Use -fdebug-prefix-map=$(ROOT_DIR)=wasisdk://v$(VERSION)"
 
 check:
-	CC="clang --sysroot=$(BUILD_PREFIX)/share/wasi-sysroot" \
-	CXX="clang++ --sysroot=$(BUILD_PREFIX)/share/wasi-sysroot -fno-exceptions" \
-	PATH="$(PATH_PREFIX)/bin:$$PATH" tests/run.sh "$(BUILD_PREFIX)" "$(RUNTIME)" "$(ADAPTER)" "$(WASM_TOOLS)"
+	TARGETS="$(TARGETS)" tests/run.sh "$(BUILD_PREFIX)" "$(RUNTIME)" "$(ADAPTER)" "$(WASM_TOOLS)"
 
 clean:
 	rm -rf build $(DESTDIR)
 
+# Default symlinks that clang creates to the `clang` executable
+CLANG_LINKS_TO_CREATE = clang++ clang-cl clang-cpp
+
+# Add target-prefixed versions of `clang` and `clang++` so they can be used
+# without `--target` as it's auto-inferred from the executable name by clang.
+CLANG_LINKS_TO_CREATE += $(foreach target,$(TARGETS),$(target)-clang)
+CLANG_LINKS_TO_CREATE += $(foreach target,$(TARGETS),$(target)-clang++)
+
+# Small helper to create a `join-with` function that can join elements of a
+# list with a defined separator.
+noop =
+space = $(noop) $(noop)
+join-with = $(subst $(space),$1,$(strip $2))
```

Comment:
Shame that we have to resort for this kind of Makefile magic, but I am not well versed enough to suggest anything better :)

Reply:
Agreed :(

I went through a number of iterations of "copy some bits and pieces from stack overflow" until I found this which ended up working well enough.